### PR TITLE
[css-pseudo] rework highlight currentColor tests

### PR DESCRIPTION
**Pull request by delan**
**(see also chromium [CL:3781986](https://crrev.com/c/3781986))**

There are many currentColor-related highlight pseudo tests, to the point where they would be easier to navigate if reorganised into their own suite. This patch renames them under “highlight-currentcolor” and gives them new assert metas as follows:

* **-computed** tests check getComputedStyle with all five kinds of highlight pseudos
    * -computed (was highlight-pseudos-currentcolor-computed-001) [^1]
    * -computed-inheritance (was highlight-pseudos-currentcolor-inheritance-computed-001) [^2]
    * -computed-visited (was highlight-pseudos-currentcolor-visited-computed-001) [^3]
* **-painting-properties** tests check painting with other properties also set to ‘currentColor’
    * -painting-properties-001 (was highlight-painting-currentcolor-001) [^4]
    * -painting-properties-002 (was highlight-painting-currentcolor-002) [^5]
* **-root-explicit-default** tests check painting with :root ‘color’ set to ‘currentColor’ via explicit defaulting
    * -root-explicit-default-001 (was highlight-painting-currentcolor-003) [^6]
    * -root-explicit-default-002 (was highlight-painting-currentcolor-004) [^7]
* **-root-implicit-default** tests check painting with :root ‘color’ set to ‘currentColor’ via implicit defaulting
    * -root-implicit-default-001 (was highlight-pseudos-currentcolor-inheritance-computed-002) [^8]
    * -root-implicit-default-002 (was highlight-pseudos-currentcolor-inheritance-computed-003) [^9]

This patch also updates the following tests:

* **-painting-properties-{001,002}**: removed ‘-webkit-text-fill-color’ and ‘-webkit-text-stroke-color’, as the spec doesn’t say these properties are applicable to highlight pseudos

[^1]: Checks the result of getComputedStyle for each kind of highlight pseudo, when ‘color’ and ‘background-color’ are set to ‘currentColor’. The result for both properties should equal the originating element ‘color’.
[^2]: Checks the result of getComputedStyle for each kind of highlight pseudo, when ‘color’ and ‘background-color’ are set to ‘currentColor’ in the parent. The child should inherit both properties as if if they were directly set to ‘currentColor’, due to highlight inheritance and the fact that ‘currentColor’ computes to itself, and the result for both properties should equal the originating element ‘color’.
[^3]: Checks the result of getComputedStyle for each kind of highlight pseudo, when ‘color’ is set to ‘currentColor’ for visited links (and for unvisited links, via implicit defaulting up to the root). The result for both links should equal the originating element ‘color’ as if the link was unvisited.
[^4]: Checks the painting for highlight pseudos (using ::highlight), when other properties are set to ‘currentColor’ (and ‘color’ is also ‘currentColor’, via implicit defaulting up to the root). The color of each property should equal the originating element ‘color’.
[^5]: Checks the painting for highlight pseudos (using ::highlight), when other properties are set to ‘currentColor’ (and ‘color’ is also ‘currentColor’, via implicit defaulting up to the root), and there is another active highlight below. The color of each property should equal the underlying highlight ‘color’.
[^6]: Checks the painting for highlight pseudos (using ::highlight), when ‘color’ is explicitly defaulted in a universal rule. ‘inherit’ (and ‘unset’, due to highlight inheritance) should equal the originating element ‘color’, because ‘color’ inherits through ancestor highlight pseudos up to the root, where the inherited value is defined as ‘currentColor’ for highlights. ‘initial’ should equal the initial value of ‘color’, which is ‘CanvasText’.
[^7]: Checks the painting for highlight pseudos (using ::highlight), when ‘color’ is universally set to ‘inherit’. When the parent is explicitly set to ‘blue’, inheritance stops there, yielding blue. Otherwise inheritance reaches the root element, where the inherited value is defined as ‘currentColor’ for highlights, yielding green.
[^8]: Checks the painting for ::selection and ::highlight, when ‘background-color’ is set in the root element but ‘color’ is implicitly defaulted. ‘background-color’ suppresses the UA default ‘color’ for ::selection via paired cascade. The resultant ‘color’ inherits through ancestor highlight pseudos up to the root, where the inherited value is defined as ‘currentColor’ for highlights, yielding green.
[^9]: Checks the painting for ::selection and ::highlight, when ‘background-color’ is set in the target element but ‘color’ is implicitly defaulted. ‘background-color’ suppresses the UA default ‘color’ for ::selection via paired cascade. The resultant ‘color’ inherits through ancestor highlight pseudos up to the root, where the inherited value is defined as ‘currentColor’ for highlights, yielding green.

### CODE REVIEW

File: css/css-pseudo/highlight-currentcolor-painting-properties-001-ref.html [hunk 1]
```diff
@@ -25,13 +25,6 @@
 #textshadow-currentcolor > span {
   text-shadow: currentcolor 2px 2px;
 }
-#textfillcolor-currentcolor > span {
-  -webkit-text-fill-color: currentcolor;
-}
-#textstrokecolor-currentcolor > span {
-  -webkit-text-stroke-width: 1px;
-  -webkit-text-stroke-color: currentcolor;
-}
```
Comment (mrego) [inline css/css-pseudo/highlight-currentcolor-painting-properties-001-ref.html:34]: The spec mentions `fill-color` and others at: https://drafts.csswg.org/css-pseudo/#highlight-styling
Why are we removing them?

**Review by mrego — COMMENTED** [review]
The renaming looks good, but I've a question about the removed properties.

File: css/css-pseudo/highlight-currentcolor-painting-properties-001-ref.html (hunk 1 above)
Comment (delan) [reply css/css-pseudo/highlight-currentcolor-painting-properties-001-ref.html:34]: [‘fill-color’](https://drafts.fxtf.org/fill-stroke-3/#fill-color) and [‘stroke-color’](https://drafts.fxtf.org/fill-stroke-3/#stroke-color) are SVG-only properties that are unrelated to [‘-webkit-text-fill-color’](https://compat.spec.whatwg.org/#the-webkit-text-fill-color) and [‘-webkit-text-stroke-color’](https://compat.spec.whatwg.org/#the-webkit-text-stroke-color), even though they work roughly the same way. see also:

* https://svgwg.org/svg2-draft/painting.html#ColorProperty
* https://lists.w3.org/Archives/Public/www-style/2016Apr/0383.html

Comment (mrego) [reply css/css-pseudo/highlight-currentcolor-painting-properties-001-ref.html:34]: Ok, I got confused about that. Thanks.

**Review by mrego — APPROVED** [review]

**Pull request merged**